src/elements/file-selector/file-selector.ts

```diff
@@ -69,15 +67,37 @@ export class SbbFileSelectorElement extends SbbDisabledMixin(
   /** This will be forwarded as aria-label to the native input element. */
   @property({ attribute: 'accessibility-label' }) public accessibilityLabel: string | undefined;
 
+  /** The path of the first selected file. '' no file is selected */
+  @property({ attribute: false })
+  public override set value(value: string | null) {
+    this._hiddenInput.value = value ?? '';
+
+    if (!value) {
+      this.files = [];
+    }
+  }
+  public override get value(): string | null {
+    return this._hiddenInput?.value;
+  }
+
   /**
    * The list of selected files.
    */
-  public get files(): File[] {
-    return this.value || [];
+  @property({ attribute: false })
+  public set files(value: File[]) {
+    this._files = value ?? [];
+
+    // update the inner input
+    const dt: DataTransfer = new DataTransfer();
+    this.files.forEach((e: File) => dt.items.add(e));
+    this._hiddenInput.files = dt.files;
+
+    this.updateFormValue();
   }
-  private set _files(value: File[]) {
-    this.value = value;
+  public get files(): File[] {
+    return this._files;
   }
+  private _files: File[] = [];
```

Comment:
Maybe I didn't understand during the call, but is it really possible to hold the files as field? wouldn't it be sufficient to return this._hiddenInput.files in the getter?